### PR TITLE
Document prometheus & portal configuration

### PR DESCRIPTION
**Pull request by goruha**
## What
* Document Portal
* Document adding custom portal tabs
* Document Prometheus operator
* Document kube-prometheus
* Document adding custom Grafana dashboards
* Document adding custom metrics

### CODE REVIEW

File: content/kubernetes-backing-services/external-dns/external-dns.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ This assumes you've followed the [Geodesic Module Usage with Terraform]({{< relr
 
 Create a file in `/conf/kops-aws-platform/external-dns.tf` with following content
```
Comment (aknysh) [inline content/kubernetes-backing-services/external-dns/external-dns.md]: with the following content

File: content/kubernetes-backing-services/external-dns/external-dns.md [hunk 2]
```diff
@@ -83,3 +83,23 @@ Add to your [Kubernetes Backing Services](/kubernetes-backing-services) Helmfile
 {{% include-code-block  title="helmfile.yaml" file="kubernetes-backing-services/external-dns/examples/external-dns-helmfile.yaml" language="yaml" %}}
 
 Then follow the instructions for running [`helmfile sync`]({{< relref "tools/helmfile.md" >}}).
+
+## Usage
+
+To leverage `external-dns`, you will need to add an annotations (e.g. `kubernetes.io/tls-acme: "true"`) to the `Ingress` resource.
+
+With these in place, then `kube-lego` will handle all e2e TLS certificate issueing and save the certificate from Let's Encrypt to a secret specificied by the `tls` config.
```
Comment (aknysh) [inline content/kubernetes-backing-services/external-dns/external-dns.md]: With these in place, `kube-lego` will handle issuing of TLS certificates from Let's Encrypt and saving them to a Kubernetes secret specified by the `tls` config parameter

File: content/kubernetes-backing-services/ingress/nginx-ingress-controller.md [hunk 3]
```diff
@@ -4,18 +4,29 @@ description: ""
...
+Run follow the instructions:
+
+* If you going to use [External DNS]({{< relref "kubernetes-backing-services/external-dns/external-dns.md" >}})
+set `NGINX_INGRESS_HOSTNAME` environment variable to domain that would be used in [external dns]({{< relref "kubernetes-backing-services/external-dns/external-dns.md#usage" >}})
+* Run command to sync `helmfile`
+
+{{% dialog type="code-block" icon="fa fa-code" title="Install ingress" %}}
 ```
-helmfile -f /conf/kops/helmfile.yaml --selector namespace=kube-system,chart=nginx-ingress sync
+chamber write kops NGINX_INGRESS_HOSTNAME ingress.us-west-2.staging.aws.popchest.io
```
Comment (osterman) [inline content/kubernetes-backing-services/ingress/nginx-ingress-controller.md]: fix domain

File: content/kubernetes-platform-services/dashboard/examples/portal-helmfile-sync.txt [hunk 4]
```diff
@@ -11,3 +11,5 @@ chamber write kops PORTAL_BRAND "Staging Cluster"
 chamber write kops PORTAL_BRAND_IMAGE_URL "https://cloudposse.com/wp-content/uploads/sites/29/2018/02/SquareLogo2.png"
 chamber write kops PORTAL_BRAND_IMAGE_FAVICON_URL "https://cloudposse.com/wp-content/uploads/sites/29/2018/04/favicon-57.png"
 chamber exec kops -- helmfile -f /conf/kops/helmfile.yaml --selector namespace=monitoring,chart=portal sync
+
+https://portal.us-west-2.staging.aws.popchest.io/oauth2/callback
```
Comment (osterman) [inline content/kubernetes-platform-services/dashboard/examples/portal-helmfile-sync.txt]: Fix domain

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+Before start read basic introduction about grafana [dashboards and panels](http://docs.grafana.org/features/panels/graph/)
+
+* Sign-in to the Kubernetes Portal. Then navigate to "Grafana".
+{{< img src="/assets/kube-prometheus-4f06031c.png" title="https://grafana.portal.us-west-2.staging.example.com" >}}
+* Login to Grafana (default credentials are `admin/admin`)
+{{< img src="/assets/kube-prometheus-f5f4472c.png" title="https://grafana.portal.us-west-2.staging.example.com/login?redirect=%2F" >}}
+{{< img src="/assets/kube-prometheus-d0d17131.png" title="Grafana homepage authentificated user" >}}
+* Create dashboard
+{{< img src="/assets/kube-prometheus-ce3e9690.png" title="https://grafana.portal.us-west-2.staging.aws.popchest.io/dashboard/new?orgId=1" >}}
```
Comment (osterman) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: Fix domain

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+[Beginners guide to building dashboards](https://www.youtube.com/watch?v=&index=7&list=PLDGkOdUX1Ujo3wHw9-z5Vo12YLqXRjzg2)
+{{< youtube sKNZMtoSHN4 >}}
+
+* [Export dashboard to JSON](http://docs.grafana.org/reference/export_import/#exporting-a-dashboard)
+
+Next steps depends of way you installed `kube-prometheus`
+
+### Installed with Master Helmfile
+
+* Put the exported content to the `/conf/kops/custom_values/kube-prometheus.grafana.dashboards.yaml`
```
Comment (osterman) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: I think I'd prefer using `/conf/kops/values/`as I don't see the need to overly namespace values.

File: content/kubernetes-backing-services/external-dns/external-dns.md [hunk 7]
```diff
@@ -83,3 +83,23 @@ Add to your [Kubernetes Backing Services](/kubernetes-backing-services) Helmfile
...
+Here are some examples:
+
+{{% include-code-block title="ingress.yaml" file="kubernetes-backing-services/tls-management/examples/kube-lego-usage-ingress.yaml" language="yaml" %}}
+
+{{% include-code-block title="values.yaml" file="kubernetes-backing-services/tls-management/examples/kube-lego-usage-values.yaml" language="yaml" %}}
+
+{{% include-code-block title="helmfile.yaml" file="kubernetes-backing-services/tls-management/examples/kube-lego-usage-helmfile.yaml" language="yaml" %}}
+
+{{% dialog type="info" icon="fa-info-circle" title="Note" %}}
+There is no unified specification for helm chart values structure. Different charts may have very different structures to values. The only way to know for sure what is supported is to refer to the chart manifests.
```
Comment (aknysh) [inline content/kubernetes-backing-services/external-dns/external-dns.md]: There is no unified specification on how to structure helm chart values. Different charts may have very different structures of the value parameters. The only way to know for sure what is supported is to refer to the chart manifests

File: content/kubernetes-backing-services/external-dns/external-dns.md [hunk 8]
```diff
@@ -83,3 +83,23 @@ Add to your [Kubernetes Backing Services](/kubernetes-backing-services) Helmfile
...
+
+Provided examples are based on the `stable/chartmuseum` chart https://github.com/kubernetes/charts/blob/master/stable/chartmuseum
```
Comment (aknysh) [inline content/kubernetes-backing-services/external-dns/external-dns.md]: The examples provided here are based on

File: content/kubernetes-backing-services/ingress/nginx-ingress-controller.md [hunk 9]
```diff
@@ -4,18 +4,29 @@ description: ""
 ---
 Nginx Ingress Controller is a type of [Ingress controller](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-controllers) that uses [ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#create-a-configmap) to store the nginx configuration.
```
Comment (aknysh) [inline content/kubernetes-backing-services/ingress/nginx-ingress-controller.md]: let's capitalize `Nginx` for consistency

File: content/kubernetes-backing-services/ingress/nginx-ingress-controller.md [hunk 10]
```diff
@@ -4,18 +4,29 @@ description: ""
...
 # Dependencies
+
 None
+
 # Install
 
 You can install `nginx-ingress` in different ways, we recomend
```
Comment (aknysh) [inline content/kubernetes-backing-services/ingress/nginx-ingress-controller.md]: recommend

File: content/kubernetes-backing-services/ingress/nginx-ingress-controller.md [hunk 11]
```diff
@@ -4,18 +4,29 @@ description: ""
...
 to use Master Helmfile.
 
 ## Install with Master Helmfile
 
-Run following command
+Run follow the instructions:
```
Comment (aknysh) [inline content/kubernetes-backing-services/ingress/nginx-ingress-controller.md]: Follow these instructions:

File: content/kubernetes-backing-services/ingress/nginx-ingress-controller.md (hunk 3 above)
Comment (aknysh) [inline content/kubernetes-backing-services/ingress/nginx-ingress-controller.md]: remove `popchest`

File: content/kubernetes-backing-services/ingress/nginx-ingress-controller.md [hunk 12]
```diff
@@ -4,18 +4,29 @@ description: ""
...
+
+* If you going to use [External DNS]({{< relref "kubernetes-backing-services/external-dns/external-dns.md" >}})
+set `NGINX_INGRESS_HOSTNAME` environment variable to domain that would be used in [external dns]({{< relref "kubernetes-backing-services/external-dns/external-dns.md#usage" >}})
```
Comment (aknysh) [inline content/kubernetes-backing-services/ingress/nginx-ingress-controller.md]: that will be used

File: content/kubernetes-backing-services/ingress/nginx-ingress-controller.md [hunk 13]
```diff
@@ -4,18 +4,29 @@ description: ""
...
+
+{{% dialog type="code-block" icon="fa fa-code" title="Install ingress" %}}
 ```
-helmfile -f /conf/kops/helmfile.yaml --selector namespace=kube-system,chart=nginx-ingress sync
+chamber write kops NGINX_INGRESS_HOSTNAME ingress.us-west-2.staging.aws.popchest.io
+chamber exec kops -- helmfile -f /conf/kops/helmfile.yaml --selector namespace=kube-system,chart=nginx-ingress sync
 ```
+{{% /dialog %}}
+
 These environment variables are used to configure:
```
Comment (aknysh) [inline content/kubernetes-backing-services/ingress/nginx-ingress-controller.md]: These environment variables are used to configure `Nginx` Ingress:

File: content/kubernetes-backing-services/ingress/nginx-ingress-controller.md [hunk 14]
```diff
@@ -4,18 +4,29 @@ description: ""
...
 
 * `NGINX_INGRESS_REPLICA_COUNT` - Count of nginx ingress pods
```
Comment (aknysh) [inline content/kubernetes-backing-services/ingress/nginx-ingress-controller.md]: `Nginx` Ingress pod replica count

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 15]
```diff
@@ -0,0 +1,134 @@
+---
+title: "Kube Prometheus"
+description: "Kube Prometheus provide easy to operate end-to-end Kubernetes cluster monitoring with Prometheus using the Prometheus Operator."
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: provides

Comment (aknysh) [reply content/kubernetes-backing-services/monitoring/kube-prometheus.md]: easy-to-operate

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+tags:
+- "grafana"
+- "prometheus"
+- "alert-manager"
+---
+
+[Kube Prometheus](https://github.com/coreos/prometheus-operator/tree/master/helm/kube-prometheus) provide easy to operate end-to-end Kubernetes cluster monitoring with Prometheus using the Prometheus Operator.
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: provides easy-to-operate

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+
+Kube Prometheus includes following packages:
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: the following

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+
+{{% dialog type="code-block" icon="fa fa-code" title="Install kube-prometheus" %}}
+```
+helmfile -f /conf/kops/helmfile.yaml --selector namespace=monitoring,chart=kube-prometheus sync
+```
+{{% /dialog %}}
+
+These are some of the environment variables you may want to configure:
+
+* `KUBE_PROMETHEUS_REPLICA_COUNT` -Count of `prometheus` pods
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: `prometheus` pod replica count

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+* `KUBE_PROMETHEUS_IMAGE_TAG` -Version of [`prometheus` image](https://quay.io/repository/prometheus/prometheus)
+* `KUBE_PROMETHEUS_ALERT_MANAGER_REPLICA_COUNT` - Count of `alertmanager` pods
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: `alertmanager ` pod replica count

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 20]
```diff
@@ -0,0 +1,134 @@
...
+{{% include-code-block  title="helmfile.yaml" file="kubernetes-backing-services/monitoring/examples/prometheus-operator-helmfile.yaml" language="yaml" %}}
+
+Then follow the instructions for running [`helmfile sync`]({{< relref "tools/helmfile.md" >}}).
+
+# Extending
+
+You can extend default functionality of `kube-prometheus` making it collect custom
+metrics, make alerts based on it and display the metric in Grafana charts.
+
+## Collection custom metrics
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: Custom metrics collection

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 21]
```diff
@@ -0,0 +1,134 @@
...
+
+Add to your [Kubernetes Backing Services](/kubernetes-backing-services) Helmfile this code
+
+{{% include-code-block  title="helmfile.yaml" file="kubernetes-backing-services/monitoring/examples/prometheus-operator-helmfile.yaml" language="yaml" %}}
+
+Then follow the instructions for running [`helmfile sync`]({{< relref "tools/helmfile.md" >}}).
+
+# Extending
+
+You can extend default functionality of `kube-prometheus` making it collect custom
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: You can extend the default functionality of `kube-prometheus` to collect custom
metrics and send alerts based on the metrics, and display the metric in Grafana charts

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 22]
```diff
@@ -0,0 +1,134 @@
...
+
+### Examples
+
+* Running Exporters with [Prometheus Operator](https://coreos.com/operators/prometheus/docs/latest/user-guides/running-exporters.html)
+* Custom configuration of [Service Monitor](https://coreos.com/operators/prometheus/docs/latest/custom-configuration.html)
+* Helm chart that exposes [Etcd metrics](https://github.com/coreos/prometheus-operator/tree/master/helm/exporter-kube-etcd)
+
+## Custom dashboards with Grafana
+
+Before start read basic introduction about grafana [dashboards and panels](http://docs.grafana.org/features/panels/graph/)
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: Before you start, read the basic

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 23]
```diff
@@ -0,0 +1,134 @@
...
+{{< img src="/assets/kube-prometheus-d0d17131.png" title="Grafana homepage authentificated user" >}}
+* Create dashboard
+{{< img src="/assets/kube-prometheus-ce3e9690.png" title="https://grafana.portal.us-west-2.staging.aws.popchest.io/dashboard/new?orgId=1" >}}
+Follow [Grafana dashboards documentations](http://docs.grafana.org/reference/templating/), read nice post [How to Create Your Own Grafana Dashboards](https://mapr.com/support/s/article/How-to-Create-Your-Own-Grafana-Dashboards?language=en_US) or watch
+[Beginners guide to building dashboards](https://www.youtube.com/watch?v=&index=7&list=PLDGkOdUX1Ujo3wHw9-z5Vo12YLqXRjzg2)
+{{< youtube sKNZMtoSHN4 >}}
+
+* [Export dashboard to JSON](http://docs.grafana.org/reference/export_import/#exporting-a-dashboard)
+
+Next steps depends of way you installed `kube-prometheus`
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: The next steps depend on the way

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md (hunk 6 above)
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: into the

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 24]
```diff
@@ -0,0 +1,134 @@
...
+    dashboard_unique_file_name.json: |-
+      EXPORTED CONTENT GOES HERE
+```
+* [Rebuild the Geodesic Module]({{< relref "geodesic/module/quickstart.md#build-install" >}})
+* [Run into the Geodesic Module shell]({{< relref "geodesic/module/quickstart.md#run-the-shell" >}})
+* Continue from step [Install with Master Helmfile]({{< relref "#install-with-master-helmfile" >}})
+
+### Installed with Custom Helmfile
+
+* Put the exported content to a `value.yaml`
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: into `value.yaml` file in this format

File: content/kubernetes-backing-services/monitoring/kube-prometheus.md [hunk 25]
```diff
@@ -0,0 +1,134 @@
...
+in format
+
+```yaml
+grafana:
+  serverDashboardFiles:
+    dashboard_unique_file_name.json: |-
+      EXPORTED CONTENT GOES HERE
+```
+
+* Ensure of using the `value.yaml` in the custom `helmfile`
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/kube-prometheus.md]: Make sure to use the `value.yaml` file in the

File: content/kubernetes-backing-services/monitoring/prometheus-operator.md [hunk 26]
```diff
@@ -1,5 +1,71 @@
 ---
 title: "Prometheus Operator"
 description: ""
-draft: true
+draft:
 ---
+[Prometheus operator](https://github.com/coreos/prometheus-operator) provide
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/prometheus-operator.md]: provides

File: content/kubernetes-backing-services/monitoring/prometheus-operator.md [hunk 27]
```diff
@@ -1,5 +1,71 @@
...
+[CRD](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
+that simplify creation/configuration/managment of [Prometheus]({{< relref "monitoring-and-alerting/prometheus.md" >}}) and [AlertManager]({{< relref "monitoring-and-alerting/alert-manager.md" >}}).
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/prometheus-operator.md]: simplifies

File: content/kubernetes-backing-services/monitoring/prometheus-operator.md [hunk 28]
```diff
@@ -1,5 +1,71 @@
...
+
+Add to your [Kubernetes Backing Services](/kubernetes-backing-services) Helmfile this code
+
+{{% include-code-block  title="helmfile.yaml" file="kubernetes-backing-services/monitoring/examples/prometheus-operator-helmfile.yaml" language="yaml" %}}
+
+Then follow the instructions for running [`helmfile sync`]({{< relref "tools/helmfile.md" >}}).
+
+# Usage
+
+Prometheus operator provide new Kubernetes resources:
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/prometheus-operator.md]: provides these new

File: content/kubernetes-backing-services/monitoring/prometheus-operator.md [hunk 29]
```diff
@@ -1,5 +1,71 @@
...
+
+* Prometheus
+* ServiceMonitor
+* Alertmanager
+
+This resources can be configured to interact with each other.
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/prometheus-operator.md]: These

File: content/kubernetes-backing-services/monitoring/prometheus-operator.md [hunk 30]
```diff
@@ -1,5 +1,71 @@
...
+
+{{% dialog type="tip" icon="fa fa-hand-o-right" title="Tip" %}}
+We recommend to install [kube-prometheus]({{< relref "kubernetes-backing-services/monitoring/kube-prometheus.md" >}})
+that install Prometheus, Alert Manager and ServiceMonitors+Exporters to
```
Comment (aknysh) [inline content/kubernetes-backing-services/monitoring/prometheus-operator.md]: installs

File: content/kubernetes-platform-services/dashboard/cluster-portal.md [hunk 31]
```diff
@@ -56,14 +56,30 @@ uses GitHub OAuth provider and is configured to expose the following dashboards:
...
+* [Alert Manager]({{< relref "kubernetes-backing-services/monitoring/kube-prometheus.md" >}})
   - accessible at `https://alerts.portal.us-west-2.staging.example.com`
-* [Grafana]({{< relref "kubernetes-backing-services/monitoring/prometheus-alerts-grafana.md" >}})
+* [Grafana]({{< relref "kubernetes-backing-services/monitoring/kube-prometheus.md" >}})
   - accessible at `https://grafana.portal.us-west-2.staging.example.com`
 * [External Documentation](https://docs.cloudposse.com)
 
+#### Adding additional tabs
+
+If you want to add some additional tabs follow this instructions
```
Comment (aknysh) [inline content/kubernetes-platform-services/dashboard/cluster-portal.md]: tabs, follow these instructions

File: content/kubernetes-platform-services/dashboard/cluster-portal.md [hunk 32]
```diff
@@ -98,15 +114,15 @@ Use these environment variables to configure the backends:
 * [Kubernetes Dashboard]({{< relref "kubernetes-platform-services/dashboard/kubernetes-ui-dashboard.md" >}})
   - `PORTAL_BACKEND_K8S_DASHBOARD_NAME` - menu item name for [kubernetes dashboard]({{< relref "kubernetes-platform-services/dashboard/kubernetes-ui-dashboard.md" >}})
   - `PORTAL_BACKEND_K8S_DASHBOARD_ENDPOINT` - internal endpoint to [kubernetes dashboard]({{< relref "kubernetes-platform-services/dashboard/kubernetes-ui-dashboard.md" >}})
```
Comment (aknysh) [inline content/kubernetes-platform-services/dashboard/cluster-portal.md]: internal endpoint for

File: content/kubernetes-platform-services/dashboard/cluster-portal.md [hunk 33]
```diff
@@ -98,15 +114,15 @@ Use these environment variables to configure the backends:
...
-  - `PORTAL_BACKEND_GRAFANA_NAME` - menu item name for [grafana]({{< relref "kubernetes-backing-services/monitoring/prometheus-alerts-grafana.md" >}})
-  - `PORTAL_BACKEND_GRAFANA_ENDPOINT` - internal endpoint for [grafana]({{< relref "kubernetes-backing-services/monitoring/prometheus-alerts-grafana.md" >}})
+* [Prometheus]({{< relref "kubernetes-backing-services/monitoring/kube-prometheus.md" >}})
+  - `PORTAL_BACKEND_PROMETHEUS_NAME` - menu item name for [prometheus ui]({{< relref "kubernetes-backing-services/monitoring/kube-prometheus.md" >}})
+  - `PORTAL_BACKEND_PROMETHEUS_ENDPOINT` - internal endpoint for [prometheus ui]({{< relref "kubernetes-backing-services/monitoring/kube-prometheus.md" >}})
+* [Alert Manager]({{< relref "kubernetes-backing-services/monitoring/kube-prometheus.md" >}})
+  - `PORTAL_BACKEND_ALERTS_NAME` - menu item name for [alert manager ui]({{< relref "kubernetes-backing-services/monitoring/kube-prometheus.md" >}})
+  - `PORTAL_BACKEND_ALERTS_ENDPOINT` - internal endpoint [alert manager ui]({{< relref "kubernetes-backing-services/monitoring/kube-prometheus.md" >}})
```
Comment (aknysh) [inline content/kubernetes-platform-services/dashboard/cluster-portal.md]: internal endpoint for

File: content/monitoring-and-alerting/prometheus.md [hunk 34]
```diff
@@ -0,0 +1,43 @@
+---
+title: "Prometheus"
+description: "Prometheus is monitoring system and time-series database that works together with Grafana and Alert Manager to provide an end-to-end monitoring platform for Kubernetes."
```
Comment (aknysh) [inline content/monitoring-and-alerting/prometheus.md]: is a monitoring system

File: content/monitoring-and-alerting/prometheus.md [hunk 35]
```diff
@@ -0,0 +1,43 @@
...
+Othertimes, software exposes metrics in a prometheus-compatable format out-of-the-box. For a list of software that integrates natively with Prometheus, check out the [list of software exposing prometheus metrics](https://prometheus.io/docs/instrumenting/exporters/#software-exposing-prometheus-metrics)
+
+## Collecting Custom Metrics
+
+To collect custom metrics a few things need to happen:
+
+* Create an exporter service to expose metrics to Prometheus
+* Register the new Exporter service with Prometheus
+
+On practice we use [Prometheus Operator]({{< relref "kubernetes-backing-services/monitoring/prometheus-operator.md" >}}) to run and configure prometheus, so follow this [documentation]({{< relref "kubernetes-backing-services/monitoring/kube-prometheus.md" >}}) to get practical
```
Comment (aknysh) [inline content/monitoring-and-alerting/prometheus.md]: In practice

Comment (aknysh) [reply content/monitoring-and-alerting/prometheus.md]: let's capitalize `Prometheus` for consistency

File: content/kubernetes-backing-services/ingress/nginx-ingress-controller.md [hunk 36]
```diff
@@ -2,25 +2,36 @@
...
-Nginx Ingress Controller is a type of [Ingress controller](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-controllers) that uses [ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#create-a-configmap) to store the nginx configuration.
+Nginx Ingress Controller is a type of [Ingress controller](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-controllers) that uses [ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#create-a-configmap) to store the Nginx configuration.
 # Dependencies
+
 None
+
 # Install
 
-You can install `nginx-ingress` in different ways, we recomend
+You can install `Nginx-ingress` in different ways, we recommend
```
Comment (aknysh) [inline content/kubernetes-backing-services/ingress/nginx-ingress-controller.md]: let's not capitalize the module and chart names

File: content/kubernetes-backing-services/ingress/nginx-ingress-controller.md [hunk 37]
```diff
@@ -29,7 +40,7 @@ Environment variables can be specified in Geodesic Module `Dockerfile` or in [Ch
 
 Add to your [Kubernetes Backing Services](/kubernetes-backing-services) Helmfile this code
 
-{{% include-code-block  title="helmfile.yaml" file="kubernetes-backing-services/ingress/examples/nginx-ingess-helmfile.yaml" language="yaml" %}}
+{{% include-code-block  title="helmfile.yaml" file="kubernetes-backing-services/ingress/examples/Nginx-ingess-helmfile.yaml" language="yaml" %}}
```
Comment (aknysh) [inline content/kubernetes-backing-services/ingress/nginx-ingress-controller.md]: let's not capitalize names in URLs

File: content/kubernetes-backing-services/ingress/nginx-ingress-controller.md [hunk 38]
```diff
@@ -39,14 +50,14 @@ After install you the ingress controller, then you can create [Ingress Resources
...
 Here are some examples:
 
-{{% include-code-block title="ingress.yaml" file="kubernetes-backing-services/ingress/examples/nginx-ingress-usage-kubectl-resource.yaml" language="yaml" %}}
+{{% include-code-block title="ingress.yaml" file="kubernetes-backing-services/ingress/examples/Nginx-ingress-usage-kubectl-resource.yaml" language="yaml" %}}
 
-{{% include-code-block title="values.yaml" file="kubernetes-backing-services/ingress/examples/nginx-ingress-usage-helm-values.yaml" language="yaml" %}}
+{{% include-code-block title="values.yaml" file="kubernetes-backing-services/ingress/examples/Nginx-ingress-usage-helm-values.yaml" language="yaml" %}}
 
-{{% include-code-block title="helmfile.yaml" file="kubernetes-backing-services/ingress/examples/nginx-ingress-usage-helmfile.yaml" language="yaml" %}}
+{{% include-code-block title="helmfile.yaml" file="kubernetes-backing-services/ingress/examples/Nginx-ingress-usage-helmfile.yaml" language="yaml" %}}
```
Comment (aknysh) [inline content/kubernetes-backing-services/ingress/nginx-ingress-controller.md]: let's not capitalize names in URLs

**Pull request merged**